sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -620,3 +621,66 @@ object CollationCheck extends (LogicalPlan => Unit) {
   private def isCollationExpression(expression: Expression): Boolean =
     expression.isInstanceOf[Collation] || expression.isInstanceOf[Collate]
 }
+
+
+/**
+ * This rule checks for references to views WITH SCHEMA [TYPE] EVOLUTION and synchronizes the
+ * catalog if evolution was detected.
+ * It does so by walking the resolved plan looking for View operators for persisted views.
+ */
+object SyncViewsCheck extends (LogicalPlan => Unit) {
+    def apply(plan: LogicalPlan): Unit = {
+    plan.foreach {
+      case View(metaData, isTempView, viewQuery)
+        if (metaData.viewSchemaMode != SchemaUnsupported && !isTempView) =>
+        val viewSchemaMode = metaData.viewSchemaMode
+        if (viewSchemaMode == SchemaTypeEvolution || viewSchemaMode == SchemaEvolution) {
+          val viewFields = metaData.schema.fields
+          val planFields = plan.schema.fields
```

Comment:
why do we care `plan.schema`? It's an arbitrary plan that reads a view.

Reply:
I've replaced plan with viewQuery.
I think these are the same though, right? view() just passes the schema through